### PR TITLE
teleport 17.0.1 (new cask)

### PR DESCRIPTION
**Pull request by daeho-ro**
add new cask for tsh alternative

----

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/t/teleport.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  desc "Modern SSH server for teams managing distributed infrastructure"
+  homepage "https://goteleport.com/"
+
+  livecheck do
+    url "https://goteleport.com/download/"
+    regex(/teleport[._-]v?(\d+(?:\.\d+)+)\.pkg/i)
+  end
+
+  conflicts_with cask:    ["tsh@13", "tsh"],
+                 formula: "teleport"
```
Comment (SMillerDev) [inline Casks/t/teleport.rb:17]: This is only on x86 right? Do we have a way to define that?

Comment (daeho-ro) [reply Casks/t/teleport.rb:17]: Cask is universal and formula also support both.

Comment (SMillerDev) [reply Casks/t/teleport.rb:17]: Sure, but it doesn't conflict with the formula on ARM since everything installs in `/opt/homebrew` there.

Comment (daeho-ro) [reply Casks/t/teleport.rb:17]: Oh, I got it. Only conflicts with formula for arm.

Comment (SMillerDev) [reply Casks/t/teleport.rb:17]: I guess there is no way to do this then

**Review by SMillerDev — APPROVED** [review]

File: Casks/t/teleport.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (khipp) [inline Casks/t/teleport.rb:17]:
```suggestion
  conflicts_with cask:    [
                   "tsh",
                   "tsh@13",
                 ],
                 formula: "teleport"
```

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**